proto/dydxprotocol/clob/query.proto

```diff
@@ -220,3 +220,36 @@ message StreamOrderbookFill {
   // Resulting fill amounts for each order in the orders array.
   repeated uint64 fill_amounts = 3;
 }
+
+// StreamTakerOrder provides information on a taker order that was attempted
+// to be matched on the orderbook.
+// It is intended to be used only in full node streaming.
+message StreamTakerOrder {
+  // The taker order that was matched on the orderbook. Can be a
+  // regular order or a liquidation order.
+  oneof taker_order {
+    Order order = 1;
+    StreamLiquidationOrder liquidation_order = 2;
+  }
+
+  // Information on the taker order after it is matched on the book,
+  // either successfully or unsuccessfully.
+  StreamTakerOrderStatus taker_order_status = 3;
+}
+
+// StreamTakerOrderStatus is a representation of match information for a  taker order
+// after it is attempted to be matched on the orderbook.
+// It is intended to be used only in full node streaming.
+message StreamTakerOrderStatus {
+  // The state of the taker order after attempting to match it against the orderbook.
+  uint32 order_status = 1;
+
+  // The amount of remaining (non-matched) base quantums of this taker order.
+  uint64 remaining_quantums = 2;
+
+	// The amount of base quantums that were optimistically filled (from this current
+  // matching cycle) of this taker order. Note that if any quantums of this order
```

Comment:
what's a current matching cycle? a proposed block?

Reply:
This wording is a little confusing to read imo. Perhaps:

// The amount of base quantums that were optimistically filled for this taker order during the proposed block. This doesn't include prior fills.

Reply:
Hmm that statement is incorrect since this is one invocation of the  matching loop (in checkTx, optimistic matching). Processing a proposed block is in DeliverTx and we just record matches instead of placing orders. Updated the comment